### PR TITLE
[BUGFIX] Show error messages in multistep checkout

### PR DESCRIPTION
**Pull request by rintisch**
Due to the implementation of the PRG pattern the
errors directly next to the errorneous fields were no longer displayed.

With this commit the PRG pattern will only be
used if no errors occur during the current step.

Fixes: #532

### CODE REVIEW

File: Classes/Controller/Cart/CartController.php [hunk 1]
```diff
@@ -125,7 +126,17 @@ public function showAction(
             $currentStep = (int)$this->request->getArgument('step');
         }
 
-        if ($currentStep && $this->request->getMethod() === 'POST') {
+        $currentStepHasError = false;
+        if ($extbaseAttribute instanceof ExtbaseRequestParameters
+            && $extbaseAttribute->getOriginalRequestMappingResults()
+            && $extbaseAttribute->getOriginalRequestMappingResults()->hasErrors()) {
```
Comment (extcode) [inline Classes/Controller/Cart/CartController.php:132]: Little code style improvement to easier catch that the condition block is closed:
```suggestion
            && $extbaseAttribute->getOriginalRequestMappingResults()->hasErrors()
        ) {
```

Comment (rintisch) [reply Classes/Controller/Cart/CartController.php]: Done.

**Comment by rintisch** [comment]
@extcode The failing test is out of scope of my changes. Could you have a look at it?

**Comment by extcode** [comment]
> @extcode The failing test is out of scope of my changes. Could you have a look at it?

This was an issue with the new version 8.1.0 of the typo3/testing-framework. It was fixed today with version 8.1.1. I restarted the faild job.

**Review by extcode — APPROVED** [review]

**Pull request merged**